apps/mobile/src/components/notifications/use-notifications.ts

```diff
@@ -0,0 +1,106 @@
+import { useCallback } from 'react';
+import { PermissionsAndroid, Platform } from 'react-native';
+
+import { useToastContext } from '@/components/toast/toast-context';
+import { useRegisterNotificationMutation } from '@/queries/notifications/notifications.query';
+import { useBitcoinAccounts } from '@/store/keychains/bitcoin/bitcoin-keychains.read';
+import { useStacksSignerAddresses } from '@/store/keychains/stacks/stacks-keychains.read';
+import { featureFlagKeys } from '@/utils/feature-flag';
+import { useBoolVariation } from '@launchdarkly/react-native-client-sdk';
+import messaging from '@react-native-firebase/messaging';
+import * as Notifications from 'expo-notifications';
+
+interface RegisterNotifications {
+  /**
+   * TODO: Unregister FCM token. New notifications sent to this token would fail.
+   * We should use this endpoint when user resets the app (deletes all wallets)
+   *
+   */
+  unregisterNotifications(): Promise<void>;
+
+  /** Registers notifications for BTC and STX addresses in our backend.
+   *
+   * /register endpoint is going to work as a sync for addresses connected to this device (across multiple wallets).
+   * We need to call it when user:
+   * 1.Adds an account
+   * 2.Adds a wallet
+   * 3.Deletes a wallet
+   * 4. Toggles the notification setting
+   *
+   */
+  registerNotifications(): Promise<void>;
+
+  requestUserPermissions(): Promise<void>;
+}
+
+export function useRegisterNotifications(): RegisterNotifications {
+  const keychains = useBitcoinAccounts();
+  const { mutateAsync: registerToken } = useRegisterNotificationMutation();
+  const stacksAddresses = useStacksSignerAddresses();
+  const { displayToast } = useToastContext();
+  const isNotificationsEnabled = useBoolVariation(featureFlagKeys.notifications, false);
```

Comment:
IMO this should be more encapsulated, eg we should export feature flag hooks from a directory so that each view doesn't have to know about `useXVariation`, it just calls `usePushNotificationsEnabled()`. Perhaps even better if the feature flags constant exports the flag name and the default value.

Reply:
`useFeatureFlags` would make a lot of sense and be quite a DX improvement (assuming type safety and default value funcitonality is preserved), but there's likely a reason LD implemented `useFlags` for the React web SDK, but omitted it for RN. They [mention it being deliberate](https://docs.launchdarkly.com/sdk/features/evaluating#react-native), but don't give any rationale. 
I can only assume this has to do with caching, and avoiding too many rerenders as RN is much more sensitive to it.
Looks like the SDK is open source, I can try and read through it later today to understand the rationale.

Reply:
P.S.
To Will's point, it would make sense to at least extract to hooks like `usePushNotificationsEnabled`, both for reausability and less boilerplate. If anything we can look into having a one-hook-to-rule-it-all later.

Reply:
Just saw this resolved thread after leaving another review. It doesn't sound like a single `useFeatureFlags` hook really does much for us _and_ there is potentially that it is hard to deal with or non-performant so it could be a pain. In that case, since there isn't any real clear advantage and many potential drawbacks I think we go with individual hooks. That's also more inline with every other implementation I've used.